matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt

```diff
@@ -411,9 +426,57 @@ internal class TimelineChunk(
     private fun buildTimelineEvent(eventEntity: TimelineEventEntity) = timelineEventMapper.map(
             timelineEventEntity = eventEntity,
             buildReadReceipts = timelineSettings.buildReadReceipts
-    ).let {
+    ).let { timelineEvent ->
         // eventually enhance with ui echo?
-        (uiEchoManager?.decorateEventWithReactionUiEcho(it) ?: it)
+        uiEchoManager?.decorateEventWithReactionUiEcho(timelineEvent)?.let { timelineEventWithEcho ->
```

Comment:
This code is a bit hard to read...

Reply:
i can create a val with the result if you prefer ?

Reply:
Maybe we can create an enhance delegation method, so we could add more in the future too